### PR TITLE
MDP-189: Hamburger menu

### PR DESCRIPTION
**Pull request by PavolKiss**

### CODE REVIEW

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
 
-export const Header = () => (
-  <div>
-    <div className="flex justify-between items-center content-center h-16 lg:h-12 flex justify-between items-center content-center w-full">
-      <Menu />
-      <ImageHeader src={chatImg} width={73} className="h-6 lg:ml-12" />
-      <Navigation />
-      <UserProfile />
+export const Header = () => {
+  const [openMenu, setOpenMenu] = useState("closed");
```
Comment (pggl) [inline src/components/Header/Header.tsx]: why you use "closed" and "open" and not boolean for this

File: src/components/Header/ImageHeader.tsx [hunk 2]
```diff
@@ -3,9 +3,13 @@ import React, { FC } from "react";
...
+  onClick?: () => void;
 }
 
-export const ImageHeader: FC<IProps> = ({ src, width = "auto", className }) => (
-  <img src={src} width={width} className={className} />
-);
+export const ImageHeader: FC<IProps> = ({
+  src,
+  width = "auto",
+  className = "",
```
Comment (pggl) [inline src/components/Header/ImageHeader.tsx]: use classnames library, it is already in master branch

File: src/components/Header/ImageHeader.tsx [hunk 3]
```diff
@@ -3,9 +3,13 @@ import React, { FC } from "react";
...
+  onClick
```
Comment (pggl) [inline src/components/Header/ImageHeader.tsx]: do we need this here?

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/components/Header/ImageHeader.tsx [hunk 4]
```diff
@@ -1,11 +1,12 @@
 import React, { FC } from "react";
+import classnames from "classnames";
```
Comment (garySiladi) [inline src/components/Header/ImageHeader.tsx:2]: we don't need the `classnames` package in this file, as the prop `className` is the only source of class here.

File: src/components/Header/ImageHeader.tsx [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
 interface IProps {
   src: string;
   width?: number;
-  className: string;
+  className?: string;
 }
 
 export const ImageHeader: FC<IProps> = ({ src, width = "auto", className }) => (
-  <img src={src} width={width} className={className} />
+  <img src={src} width={width} className={classnames(className)} />
```
Comment (garySiladi) [inline src/components/Header/ImageHeader.tsx:11]: this should be then `className={className}`

File: src/components/Header/ImageHeader.tsx [hunk 6]
```diff
@@ -1,11 +1,12 @@
...
+import classnames from "classnames";
 
 interface IProps {
   src: string;
   width?: number;
-  className: string;
+  className?: string;
 }
 
 export const ImageHeader: FC<IProps> = ({ src, width = "auto", className }) => (
```
Comment (garySiladi) [inline src/components/Header/ImageHeader.tsx:10]: set empty string as default value of className

File: src/components/Header/Header.tsx [hunk 7]
```diff
@@ -1,18 +1,37 @@
...
-    <div className="flex justify-between items-center content-center h-16 lg:h-12 flex justify-between items-center content-center w-full">
-      <Menu />
-      <ImageHeader src={chatImg} width={73} className="h-6 lg:ml-12" />
-      <Navigation />
-      <UserProfile />
+export const Header = () => {
+  const [openMenu, setOpenMenu] = useState(false);
+  return (
+    <div>
+      {openMenu === true && (
```
Comment (garySiladi) [inline src/components/Header/Header.tsx]: you can check if `openMenu` is explicitly `true` just with the expression
```
{openMenu && (...
```

File: src/static/svg/times.svg [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24">
```
Comment (garySiladi) [inline src/static/svg/times.svg]: why is this image called `times` if it's a close button?

File: src/components/Header/Header.tsx [hunk 9]
```diff
@@ -1,18 +1,37 @@
...
+          </div>
+          <HamburgerMenu />
+          <div
+            className="opacity-75 bg-darkblue h-full w-full lg:hidden"
+            onClick={() => setOpenMenu(false)}
+          />
+        </div>
+      )}
+      <div className="flex justify-between items-center content-center h-16 lg:h-12 w-full">
+        <Menu onClick={() => setOpenMenu(true)} />
```
Comment (garySiladi) [inline src/components/Header/Header.tsx:29]: as I tried this in browser, I would like to have pointer cursor and some hover effect when I hover over this icon from a design standpoint

File: src/components/Header/Header.tsx [hunk 10]
```diff
@@ -1,18 +1,37 @@
...
+      {openMenu === true && (
+        <div>
+          <div className="flex justify-end mr-6 mt-4">
+            <div onClick={() => setOpenMenu(false)}>
+              <ImageHeader src={times} />
+            </div>
+          </div>
+          <HamburgerMenu />
+          <div
+            className="opacity-75 bg-darkblue h-full w-full lg:hidden"
```
Comment (garySiladi) [inline src/components/Header/Header.tsx:23]: is the darkblue showing for you in the browser? for me it's white

**Review by garySiladi — CHANGES_REQUESTED** [review]

**Review by garySiladi — APPROVED** [review]
ok from my side

**Review by pggl — APPROVED** [review]

**Pull request merged**